zebra-consensus/src/block/check.rs

```diff
@@ -51,6 +56,39 @@ pub fn is_equihash_solution_valid(header: &Header) -> Result<(), equihash::Error
 /// accepted." [§7.5][7.5]
 ///
 /// [7.5]: https://zips.z.cash/protocol/protocol.pdf#blockheader
-pub fn is_time_valid_at(header: &Header, now: DateTime<Utc>) -> Result<(), BoxError> {
+pub fn time_is_valid_at(header: &Header, now: DateTime<Utc>) -> Result<(), BoxError> {
     header.is_time_valid_at(now)
 }
+
+/// [3.9]: https://zips.z.cash/protocol/protocol.pdf#subsidyconcepts
+pub fn subsidy_is_correct(network: Network, block: &Block) -> Result<(), BoxError> {
+    let height = block
+        .coinbase_height()
+        .expect("always called on blocks with a coinbase height");
+
+    let coinbase = block.transactions.get(0).ok_or("no coinbase transaction")?;
+    let outputs = coinbase.outputs();
+
+    let canopy_height = Canopy.activation_height(network).ok_or("no canopy")?;
+    let last_founder_reward = (canopy_height - 1).expect("have canopy so we can substract 1");
+    if height >= canopy_height {
+        panic!("Can't validate Canopy yet");
+    }
+
+    // validate founders reward and miner subsidy
+    if height > Height(0) && height <= last_founder_reward {
+        let block_subsidy = subsidies::block_subsidy(height, Network::Mainnet) as i64;
+        let miner_subsidy = subsidies::miner_subsidy(height, Network::Mainnet) as i64;
```

Comment:
It would be good to add a test that uses the Blossom testnet activation block. That test will make sure we don't have any other similar bugs.

Also, implicitly use `Amount<NonNegative>` for currency.

```suggestion
        let block_subsidy = subsidies::block_subsidy(height, network);
        let miner_subsidy = subsidies::miner_subsidy(height, network);
```

Reply:
For the network parameter: https://github.com/ZcashFoundation/zebra/pull/1051/commits/03cdf101e2230fd5871bfcc2398b37cfcfe94e4a

Reply:
https://github.com/ZcashFoundation/zebra/pull/1051/commits/2f426ea7ab779093472f4ac633546427a678d343